src/amazon/dsstne/engine/NNNetwork.cpp

```diff
@@ -3983,41 +3985,41 @@ bool NNNetwork::P2P_Allreduce(NNFloat* pBuffer, size_t size)
-            
+                uint64_t end                            = (size * (segment + 1)) / getGpu()._numprocs;
+
                 // Send segments around the adding local contributions from each process
                 for (uint32_t i = 0; i < stages; i++)
                 {
-               
+
                     if (i == 0)
                         cudaMemcpy(GetPeerBuffer(), pBuffer + start, (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
                     else
-                        cudaMemcpy(GetPeerBuffer(), GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault); 
-                   
+                        cudaMemcpy(GetPeerBuffer(), GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
+
                     // Wait for completion
-                    cudaDeviceSynchronize();              
+                    cudaDeviceSynchronize();
                     MPI_Barrier(MPI_COMM_WORLD);
                     SwapPeerBuffers();
                     segment                             = (segment + 1) % getGpu()._numprocs;
                     start                               = (size * segment) / getGpu()._numprocs;
                     end                                 = (size * (segment + 1)) / getGpu()._numprocs;
-                    kAddBuffers(GetP2PSendBuffer(), pBuffer + start, end - start);  
+                    kAddBuffers(GetP2PSendBuffer(), pBuffer + start, end - start);
                 }
-                 
+
                 // Circulate segments a second time and copy out results
                 cudaMemcpy(pBuffer + start, GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
                 for (uint32_t i = 0; i < stages; i++)
                 {
-                    cudaMemcpy(GetPeerBuffer(), GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);   
-                   
-                    // Wait for completion  
-                    cudaDeviceSynchronize();              
+                    cudaMemcpy(GetPeerBuffer(), GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
+
+                    // Wait for completion
+                    cudaDeviceSynchronize();
                     MPI_Barrier(MPI_COMM_WORLD);
                     SwapPeerBuffers();
                     segment                             = (segment + 1) % getGpu()._numprocs;
                     start                               = (size * segment) / getGpu()._numprocs;
                     end                                 = (size * (segment + 1)) / getGpu()._numprocs;
-                    cudaMemcpy(pBuffer + start, GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);               
+                    cudaMemcpy(pBuffer + start, GetP2PSendBuffer(), (end - start) * sizeof(NNFloat), cudaMemcpyDefault);
                 }
             }
         }
```

Comment:
Completely agree that these changes would typically be split into separate CRs. I figured out that these formatting changes were accidentally checked in when I rebased my branch, as I resolved a merge conflict while also experimenting with a text editor feature that automatically trims whitespace.